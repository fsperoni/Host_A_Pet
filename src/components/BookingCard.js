import React from 'react';

const BookingCard = ({ avail }) => {
  console.log("Card", avail);
  let bookingCard;

  if (avail.role === "Host") {
    bookingCard = (
      <>
        <h2>{avail.user.username}</h2>
        <h4>{avail.user.firstName} is available on your selected dates</h4>
        <h4>Ratings: </h4>
      </>
    )
  } else {
    const pets = avail.pets.map(pet => (
      <div key={pet.id} className="col-lg mt-3 mx-3 mb-1 border border-secondary">
        <h3>{pet.name} the {pet.type}</h3>
        <img src={pet.photo} className="mt-2 img-fluid" alt="pet" />
      </div>
    ))
    bookingCard = (
      <>
        <h2 className="text-center">{avail.user.username}</h2>
        <h4 className="text-center">{avail.user.firstName} needs a host. Can you help?</h4>
        <h4>Their pets:</h4>
        <div className="row">{pets}</div>
      </>
    )
  }
  return (

    <div className="BookingCard border border-secondary">
      {bookingCard}
    </div>
  )
}

export default BookingCard;