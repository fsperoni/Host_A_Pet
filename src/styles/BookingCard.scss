.BookingCard {
  
  .alert-container {
    display: flex;
    justify-content: center;

    .alert {
      padding: 1rem 6rem;
    }
  }
}