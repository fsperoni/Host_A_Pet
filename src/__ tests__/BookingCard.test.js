import { render } from "@testing-library/react";
import BookingCard from "../components/BookingCard";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
  bookingAvail, dateRange
} = require("./_testCommon");

// smoke test
it('renders without crashing', function() {
  render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <BookingCard setBookError={function() {}} bookError={true} bookSuccess={false}
        setBookSuccess={function() {}} avail={bookingAvail()} dateRange={dateRange()}/>
    </UserContext.Provider>);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <BookingCard setBookError={function() {}} bookError={true} bookSuccess={false}
        setBookSuccess={function() {}} avail={bookingAvail()} dateRange={dateRange()}/>
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Book!' in the component.", function() {
  const { getByText } = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <BookingCard setBookError={function() {}} bookError={true} bookSuccess={false}
        setBookSuccess={function() {}} avail={bookingAvail()} dateRange={dateRange()}/>
    </UserContext.Provider>
  );
  expect(getByText("Book!")).toBeInTheDocument();
});
