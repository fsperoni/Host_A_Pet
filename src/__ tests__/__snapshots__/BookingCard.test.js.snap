// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BookingCard border border-secondary"
  >
    <h2
      class="text-center"
    >
      testuser
    </h2>
    <h4
      class="text-center"
    >
      Test needs a host. Can you help?
    </h4>
    <h4>
      Their pets:
    </h4>
    <div
      class="row"
    >
      <div
        class="col-lg mt-3 mb-1"
      >
        <h3>
          Garfield1 the Dog
        </h3>
        <img
          alt="pet"
          class="mt-2 img-fluid"
          src="/static/media/dog.4b90f018c1cda3590605.jpeg"
        />
      </div>
      <div
        class="col-lg mt-3 mb-1"
      >
        <h3>
          Garfield0 the Cat
        </h3>
        <img
          alt="pet"
          class="mt-2 img-fluid"
          src="/static/media/cat.590ca8590214640ca23b.png"
        />
      </div>
    </div>
    <div
      class="row justify-content-center my-3"
    >
      <button
        class="col-2 btn btn-primary btn-block"
      >
        <data
          value="45"
        />
        Book!
      </button>
    </div>
    <div
      class="alert-container"
    />
  </div>
</DocumentFragment>
`;
